node/src/main/scala/org/bitcoins/node/networking/P2PClient.scala

```diff
@@ -194,17 +215,24 @@ case class P2PClientActor(
           logger.trace(s"Unaligned bytes: ${newUnalignedBytes.toHex}")
         }
 
-        //for the messages we successfully parsed above
-        //send them to 'context.parent' -- this is the
-        //PeerMessageHandler that is responsible for
-        //creating this Client Actor
-        messages.foreach { m =>
-          val msg = NetworkMessageReceived(m, P2PClient(self, peer))
-          peerMsgHandlerReceiver.handleNetworkMessageReceived(msg)
-
+        val newMsgReceiverF: Future[PeerMessageReceiver] = {
+          val currentF = Future.successful(currentPeerMsgHandlerRecv)
+          logger.trace(s"About to process ${messages.length} messages")
+          messages.foldLeft(currentF) {
+            case (peerMsgRecvF, m) =>
+              peerMsgRecvF.flatMap { peerMsgRecv: PeerMessageReceiver =>
```

Comment:
Might want to consider calling `recover` on this so that one message handling failure doesn't result in other dropped messages?

Reply:
#671